src/main/java/cloudy/Parser.java

```diff
@@ -0,0 +1,191 @@
+package cloudy;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * The Parser class is responsible for interpreting user input and converting it into executable
+ * commands for the Cloudy program.
+ */
+public class Parser {
+
+    private static final String DATE_PATTERN = "^\\d{2}/\\d{2}/\\d{4}$";
+
+    /**
+     * Parses the user input into a Command object.
+     * @param userInput The user's input as a string.
+     * @return A Command object representing the user's intended action.
+     */
+    public Command parseCommand(String userInput) {
+        if (userInput.startsWith("mark ")) {
+            return parseMarkCommand(userInput);
+        } else if (userInput.startsWith("unmark ")) {
+            return parseUnmarkCommand(userInput);
+        } else if (userInput.startsWith("todo")) {
+            return parseTodoCommand(userInput);
+        } else if (userInput.startsWith("deadline")) {
+            return parseDeadlineCommand(userInput);
+        } else if (userInput.startsWith("event")) {
+            return parseEventCommand(userInput);
+        } else if (userInput.startsWith("delete")) {
+            return parseDeleteCommand(userInput);
+        } else if (userInput.startsWith("find")) {
+            return parseFindCommand(userInput);
+        } else if (userInput.equals("list")) {
+            return new Command("list");
+        } else if (userInput.equals("bye")) {
+            return new Command("bye");
+        } else {
+            return new Command("invalid");
+        }
+    }
+
+    /**
+     * Parses a mark command from the user's input.
+     * @param userInput The user's input containing the mark command.
+     * @return A command object for marking a task as complete.
+     */
+    private Command parseMarkCommand(String userInput) {
+        String[] parts = userInput.split(" ");
+        if (parts.length >= 2) {
+            try {
+                int taskNumber = Integer.parseInt(parts[1]);
+                return new Command("mark", taskNumber);
+            } catch (NumberFormatException e) {
+                return new Command("invalidTaskNum");
+            }
+        } else {
+            return new Command("invalidTaskFormat");
+        }
+    }
```

Comment:
There are repeated patterns in methods like parseMarkCommand, parseUnmarkCommand, and parseDeleteCommand. Can consider abstracting this into a helper method 👍